core/src/main/kotlin/fr/sncf/osrd/stdcm/preprocessing/implementation/BlockAvailability.kt

```diff
@@ -34,13 +36,153 @@ data class BlockAvailability(
         startOffset: Offset<Path>,
         endOffset: Offset<Path>,
         startTime: Double
+    ): BlockAvailabilityInterface.Availability {
+        var timeShift = 0.0
+        var firstConflictOffset = infraExplorer.getIncrementalPath().toTravelledPath(startOffset)
+        while (timeShift.isFinite()) {
+            val shiftedStartTime = startTime + timeShift
+            val pathStartTime =
+                shiftedStartTime - infraExplorer.interpolateDepartureFromClamp(startOffset)
+            val endTime = infraExplorer.interpolateDepartureFromClamp(endOffset) + pathStartTime
+
+            val stepAvailability =
+                getStepAvailability(
+                    infraExplorer,
+                    startOffset,
+                    endOffset,
+                    pathStartTime,
+                    shiftedStartTime
+                )
+            val conflictAvailability =
+                getConflictAvailability(
+                    infraExplorer,
+                    startOffset,
+                    pathStartTime,
+                    shiftedStartTime,
+                    endTime
+                )
+            val availability =
+                getMostRestrictiveAvailability(stepAvailability, conflictAvailability)
+
+            if (availability !is BlockAvailabilityInterface.Unavailable) {
```

Comment:
```suggestion
            if (availability is BlockAvailabilityInterface.Available) {
```

Reply:
There should be a way to make this better yeah (either that or using `when`). I'll modify this.